static/script/sketch.js

```diff
@@ -0,0 +1,356 @@
+    }
+
+    // console.log(trex.animation);
+    generate_clouds();
+    generate_cactuses();
+
+    // add a better way to make the game faster
+
+    // If player has already been playing for some time && determined frameRate is reached
+    if (score >= 300 && frameCount % 60 == 0) {
+      // Generate Pterodactylus at random heights for player to dodge
+      generate_flyingDino();
+
+      // Day and night cycle 
+      time: {
+        if (day) {
+          //Background will get clearer
+          bg -= 25;
+          //Moon will start to fade away
+          moonOpacity += 25;
+          //It will stay clear for some time...
+          setTimeout(() => {
+            if (bg <= 0) {
+              day = false;
+            }
+          }, 4000)
+          // if it's day, then the function ends here, there is no need to continue
+          break time;
+        }
+        // if it's night, then the function continues and the code inside the if statement isn't executed
+        //Background will get darker
+        bg += 25;
+        //Moon will get more visible
+        moonOpacity -= 25;
+        //It will stay dark for some time...
+        setTimeout(() => {
+          if (bg >= 156) {
+            day = true;
+          }
+        }, 4000)
+      }
+    }
+
+    // Trex dies
+    if (trex.collides(obstacleGroup) || trex.collides(flyingDinoGroup)) {
+      gameOverTxt.show();
+
+      collideSound.play();
+
+      endGame = true;
+    }
+    return
+  }
+  // bug da lua, continua andando mesmo o jogo tendo acabado ou resetado, sua posição não é resetada
+  trex.changeAnimation("dead", trexCollide);
+
+  trex.velocity.x = ground.velocity.x = 0;
+
+  // obstacles need to stay in place, otherwise player would collide with them and they would go flying away(no gravity)
+  obstacleGroup.collider = flyingDinoGroup.collider = cloudsGroup.collider = 'static';
+  obstacleGroup.life = flyingDinoGroup.life = cloudsGroup.life = Infinity;
+
+  restart.show();
+
+  document.body.addEventListener("click",  restartGame)
+  document.body.addEventListener("keydown", restartGame)
+
+  maximumScore = score > maximumScore ? score : maximumScore;
+
+  // Save player maximum score so if the page is reloaded player will still have its score record saved
+  localStorage.setItem("userRecord", maximumScore); 
+}
+
+document.addEventListener('keydown', e => {
+  switch (e.key) {
+    case "ArrowUp":
+    case "w":
+    case " ":
+      isKeyUpPressed = true;
+      break
+    case "ArrowDown":
+    case "s":
+      isKeyDownPressed = true;
+      break
+  }
+});
+
+document.addEventListener('keyup', e => {
+    switch (e.key) {
+      case "ArrowUp":
+      case "w":
+      case " ":
+        isKeyUpPressed = false;
+        break
+      case "ArrowDown":
+      case "s":
+        isKeyDownPressed = false;
+        //
+        trex.changeAnimation("trexWalk", texTrex);
+        trex.animation.offset.y = 0;
+        canJump = true;
+        break
+    }
+});

```

Comment:
this is a more independent way of keyboard events. This method does not depend of any library to work